receiver/mongodbreceiver/metrics.go

```diff
@@ -43,176 +43,154 @@ var documentMap = map[string]metadata.AttributeOperation{
 	"deleted":  metadata.AttributeOperationDelete,
 }
 
+const (
+	collectMetricError                 = "failed to collect metric %s: %w"
+	collectMetricWithAttributes        = "failed to collect metric %s with attribute(s) %s: %w"
+	collectMetricWarning               = "Failed to collect metric %s"
+	collectMetricWarningWithAttributes = "Failed to collect metric %s with attribute(s) %s"
+)
+
 // DBStats
 func (s *mongodbScraper) recordCollections(now pcommon.Timestamp, doc bson.M, dbName string, errs *scrapererror.ScrapeErrors) {
-	collectionsPath := []string{"collections"}
-	collections, err := dig(doc, collectionsPath)
-	if err != nil {
-		errs.AddPartial(1, err)
-		return
-	}
-	collectionsVal, err := parseInt(collections)
+	metricPath := []string{"collections"}
+	metricName := "mongodb.collection.count"
+	val, err := collectMetric(doc, metricPath)
 	if err != nil {
-		errs.AddPartial(1, err)
+		errs.AddPartial(1, fmt.Errorf(collectMetricWithAttributes, metricName, dbName, err))
+		s.logger.Warn(fmt.Sprintf(collectMetricWarningWithAttributes, metricName, dbName), zap.Error(err))
```

Comment:
I don't think we need both the error and the log

Reply:
Would this apply in all situations, both in scraper.go calling a client to get a payload response and for adding a metric in metrics.go? I am guessing I'll keep partial error in metrics.go file and remove the log warning. What about the in scraper.go, should those partial errors stay or should the log warn stay?